### ISSUE TITLE
[BUG] If Origin is missing from cookie, autoload js handler fails with Tornado KeyError: 'Origin'

### ISSUE THREAD

**Issue by hakan-77** [bokeh/bokeh#13987]
### Software versions

Python 3.10 Bokeh 3.5

### Browser name and version

Fails on both Chrome and Safari

### Expected behavior

The below fail when browser doesn't send Origin cookie in an embedded scenario

bokeh/server/views/autoload_js_handler.py
```python
    async def get(self, *args, **kwargs):
        if self.request.cookies:
            self.set_header("Access-Control-Allow-Origin", self.request.headers["Origin"])
```

### Observed behavior

Potential solution:

bokeh/server/views/autoload_js_handler.py
```python
    async def get(self, *args, **kwargs):
...
origin = self.request.headers.get("Origin")
if origin:
    self.set_header("Access-Control-Allow-Origin", origin)
else:
    self.set_header("Access-Control-Allow-Origin", "*") 
```

### Example code

```Python
Embedded bokeh
```

### Stack traceback or browser console output

host='...', method='GET', uri='/.../app/autoload.js?bokeh-autoload-element=c1fca0ee-4e8e-4073-90b1-53b6fc256bb9&bokeh-app-path=/.../app&bokeh-absolute-url=.../app&resources=none', version='HTTP/1.1', remote_ip='127.0.0.6')\nTraceback (most recent call last):\n  File \"/home/ubuntu/.local/lib/python3.10/site-packages/tornado/web.py\", line 1790, in _execute\n    result = await result\n  File \"/home/ubuntu/.local/lib/python3.10/site-packages/bokeh/server/views/autoload_js_handler.py\", line 63, in get\n    self.set_header(\"Access-Control-Allow-Origin\", self.request.headers[\"Origin\"])\n  File \"/home/ubuntu/.local/lib/python3.10/site-packages/tornado/httputil.py\", line 219, in __getitem__\n    return self._dict[_normalize_header(name)]\
KeyError: 'Origin'

**Comment by bryevdv** [comment]
@pmeier can you comment? I believe you made this addition in #13800

Offhand the gist of the proposed change seems OK, since that is the default value of the header anyway.  But I'd probably suggest just changing the condition:
```python
if self.request.cookies and "Origin" in self.request.headers:
```

But please advise whether this is a good thing to do or not (e.g. if there is some reason `Origin` should always be present when there are cookies, we can raise a better error). cc @bokeh/dev

**Comment by hakan-77** [comment]
@bryevdv @pmeier Your suggestion would work, no need to add "*" and I can confirm there are legit embedded cases where Origin is not part of the http header.

**Comment by bryevdv** [comment]
> I can confirm there are legit embedded cases where Origin is not part of the http header.

Can you elaborate? To be clear in this context I will need this assessment confirmed by someone other than the person making the ask cc also @philippjfr

**Comment by pmeier** [comment]
Also interested in the use case @hakan-77. In my experiments while coding the patch, I never had such a case. I can only speak for Firefox and Chromium, but not Safari.

From the [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description)

> The Origin header value may be null in a number of cases, including (non-exhaustively):
> - Origins whose scheme is not one of http, https, ftp, ws, wss, or gopher (including blob, file and data).
> - Cross-origin images and media data, including that in <img>, <video> and <audio> elements.
> - Documents created programmatically using createDocument(), generated from a data: URL, or that do not have a creator browsing context.
> - Redirects across origins.
> - iframes with a sandbox attribute that doesn't contain the value allow-same-origin.
> - Responses that are network errors.
> - [Referrer-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy) set to no-referrer for non-cors request modes (e.g. simple form posts).

Not sure which of these applies to bokeh, but I guess there are plenty of valid use cases.

As for the proposed patch, I don't if it makes sense to set any of the CORS headers if the origin is missing. So we probably don't need to set anything in that case. Or in other words, setting a partial set of CORS headers might lead to other issues or confusion down the line.

**Comment by bryevdv** [comment]
> As for the proposed patch, I don't if it makes sense to set any of the CORS headers if the origin is missing. So we probably don't need to set anything in that case. 

@pmeier just clarifying, the existing code already seems to set these by default in all cases:

https://github.com/bokeh/bokeh/blob/000d49f9bfed412be978d995ecdb88cf9d412ecf/src/bokeh/server/views/autoload_js_handler.py#L56-L59

Are you saying that should not be the case?

**Comment by hakan-77** [comment]
@pmeier We have are embedding Bokeh in an existing tornado app. Have you tried this use case? @pmeier agreed, just adding an extra check of "Origin" existing in headers is good enough.

```python
if self.request.cookies and "Origin" in self.request.headers:
```

**Comment by pmeier** [comment]
@bryevdv

> Are you saying that should not be the case?

Sorry, I was mistaken. I think the patch in https://github.com/bokeh/bokeh/issues/13987#issuecomment-2244127390 for 

https://github.com/bokeh/bokeh/blob/6404785c7f65994d0cf44fa598173661b1447e81/src/bokeh/server/views/autoload_js_handler.py#L62-L63

looks good. Maybe we can also add my GH comment https://github.com/bokeh/bokeh/pull/13800#discussion_r1551205201 as code comment for clarification.

**Comment by pmeier** [comment]
@bryevdv I've added a patch in #13999.

**Comment by bryevdv** [comment]
Fix merged in `branch-3.6` but if we ever release a 3.5.2 version it will get back ported
